modules/smart_contract_connector/lib/impl.js

```diff
@@ -16,138 +16,246 @@ const OrgId = Contracts.getFromNodeModules('@windingtree/org.id', 'OrgId');
+            resolve(false)
+        }
+    });
 
     // Get the Organizations in URL for the Domain Trust Clue
-    const getOrgidFromUrl = async (link) => {
-        return new Promise(async (resolve) => {
-            try {
-                const fetched = await fetch(`${link}/org.id`);
-                let body = await fetched.text();
-                body = body.replace('orgid=', '').replace('did:orgid:');
-                resolve(body);
-            } catch (e) {
-                resolve(false);
-            }
-        })
-    };
+    const getOrgidFromUrl = async (link) => new Promise(async (resolve) => {
+        try {
+            const fetched = await fetch(`${link}/org.id`);
+            let body = await fetched.text();
+            body = body.replace('orgid=', '').replace('did:orgid:');
+            resolve(body);
+        } catch (e) {
+            resolve(false);
+        }
+    });
 
     const getOrgidFromLink = async (link) => {
         let orgid = await getOrgidFromDns(link);
-        if (!orgid) orgid = await getOrgidFromUrl(link);
+
+        if (!orgid) {
+            orgid = await getOrgidFromUrl(link);
+        }
+
         return orgid;
     };
 
-    const checkSslByUrl = (link, expectedLegalName) => {
-        return new Promise(async (resolve) => {
-            if(link.indexOf('://') === -1) link = `https://${link}`;
-            const dns = await getOrgidFromDns(link);
-            if (dns === undefined) return resolve(dns);
-            let requestSsl;
-            try {
-                let { hostname } = new URL(link);
-                let isAuthorized = false;
-                const options = { host: hostname, method: 'get', path: '/', agent: new https.Agent({ maxCachedSessions: 0 }) };
-                let companySiteHostnameFromServer, legalNameFromServer;
-                requestSsl = https.request(options, (response) => {
-                    let subject = response.socket.getPeerCertificate().subject;
-                    let CN = subject.CN.replace('*.','');
-                    if(CN.indexOf('://') === -1) CN = `https://${CN}`;
-                    companySiteHostnameFromServer = new URL(CN).hostname;
-                    legalNameFromServer = subject.O;
-                    console.log(companySiteHostnameFromServer, legalNameFromServer);
-
-                    isAuthorized = response.socket.authorized;
-                    resolve(isAuthorized && (legalNameFromServer === expectedLegalName) && (companySiteHostnameFromServer === hostname))
-                });
-                requestSsl.end();
-            } catch (e) {
-                console.log('checkSslByUrl [ERROR]', e.toString());
-                resolve(false)
-            }
-        })
-    };
+    const checkSslByUrl = (link, expectedLegalName) => new Promise(async (resolve) => {
+       
+        if (link.indexOf('://') === -1) {
+            link = `https://${link}`;
+        }
+
+        const dns = await getOrgidFromDns(link);
```

Comment:
Same comment as above, I think the idea here is to check the organization from the SSL Certificate and not the DNS.

Reply:
I thought - we were not going to check the owner of SSL certificate.
DNS trust in ORG.ID sense is the confirmation of the existence of the TXT record with a proper organization identifier. Right? or I had mistaken?